### ISSUE TITLE
updateProps may include children

### ISSUE THREAD

**Issue by maparent** [yysun/apprun#74]
I have been doing some (admittedly complex) experiments which end up with a component receiving `updateProps` with its complete current state, which normally includes `children`. Of course, trying to assign `children` on the `htmlElement` fails [here](
https://github.com/yysun/apprun/blob/e4ef0c7658320f9ec314e862221c16a46a6f0c3b/src/vdom-my.ts#L189). Do you think `children` should be special-cased, or should I find a way to eliminate or filter `children` from the state?

P.S. I have tried eliminating `children` from the state with no adverse effect so far, and that solves my issue, but I am not sure it's the best course of action.

**Comment by yysun** [comment]
Here is the example of handling children using the _mounted_ function.
https://github.com/yysun/apprun/blob/V2/demo-html/components/counter.tsx

**Comment by maparent** [comment]
Thank you for this, it shows well how to use the children passed; but that was not my issue. I now understand where it comes from; it was arising in the example I gave of [a recursive tree](https://gist.github.com/maparent/9565cf0b504ed47ae719aca74c182617). (I updated that code to fix the bug better.)

Let me recap: When a component is rendered, the props acquire `children` [here](https://github.com/yysun/apprun/blob/5be47db699c513690377e03bc40410a4ac9bf2f0/src/createComponent.tsx#L12). My code would propagate that state in the parent's state, which would then re-propagate it to the child with `children` in the props. The child would get re-rendered, and break at the assignment of `children` to the HTML element [here](https://github.com/yysun/apprun/blob/e4ef0c7658320f9ec314e862221c16a46a6f0c3b/src/vdom-my.ts#L189), as mentioned. So it is an artefact of my strategy for state sharing; I am closing the issue. I still think it would be good to special case children in the update function, in case someone introduces that name in the props by mistake, but now that I understand where it came from, I doubt it's likely in normal usage.
Thank you.
